lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,371 @@
+                )
+
+                # generate the auth token
+                auth_token = user.encode_auth_token(user.id)
+                response_object = {
+                    "status": "success",
+                    "status_code": 201,
+                    "message": "Successfully registered.",
+                    "auth_token": auth_token.decode(),
+                }
+                return make_response(jsonify(response_object))
+
+            except Exception as e:
+                abort(401, message="Some error occurred. Please try again.")
+        else:
+            abort(409, message="User already exists. Please Log in.")
+
+
+class LoginAPI(Resource):
+    """
+    Abstracted pbench API for User Login
+    """
+
+    def __init__(self, app, bcrypt):
+        self.app = app
+        self.bcrypt = bcrypt
+
+    def post(self):
+        """
+        Post request for logging in user.
+        This requires a JSON data with required user metadata fields
+        {
+            "username": "username",
+            "password": "password",
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to decrypt the password hash, the returned response_object will include the auth_token
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully logged in."/"failure message",
+                    "auth_token": auth_token.decode(), # Will not present if failed
+                }
+        """
+        # get the post data
+        post_data = request.get_json()
+        try:
+            # fetch the user data
+            user = UserModel.query.filter_by(username=post_data.get("username")).first()
+
+            if user and self.bcrypt.check_password_hash(
+                user.password, post_data.get("password")
+            ):
+                auth_token = user.encode_auth_token(user.id)
+                if auth_token:
+                    response_object = {
+                        "status": "success",
+                        "status_code": 200,
+                        "message": "Successfully logged in.",
+                        "auth_token": auth_token.decode(),
+                    }
+                    return make_response(jsonify(response_object))
+            else:
+                abort(404, message="User does not exist.")
+        except Exception as e:
+            self.app.logger.info(f"Exception occurred during user login {e}")
+            abort(500, message="Try again")
+
+
+class LogoutAPI(Resource):
+    """
+    Abstracted pbench API for User Logout
+    """
+
+    def __init__(self, app, blacklist):
+        self.app = app
+        self.blacklist = blacklist
+
+    @jwt_required
+    def delete(self):
```

Comment:
This method needs a `try/except` to handle API and internal errors a bit more gracefully.